components/camel-ai/camel-openai/src/main/java/org/apache/camel/component/openai/OpenAIEndpoint.java

```diff
@@ -0,0 +1,128 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.camel.component.openai;
+
+import com.openai.client.OpenAIClient;
+import com.openai.client.okhttp.OpenAIOkHttpClient;
+import org.apache.camel.Category;
+import org.apache.camel.Consumer;
+import org.apache.camel.Processor;
+import org.apache.camel.Producer;
+import org.apache.camel.spi.Metadata;
+import org.apache.camel.spi.UriEndpoint;
+import org.apache.camel.spi.UriParam;
+import org.apache.camel.spi.UriPath;
+import org.apache.camel.support.DefaultEndpoint;
+
+/**
+ * OpenAI endpoint for chat completion.
+ */
+@UriEndpoint(firstVersion = "4.16.0",
+             scheme = "openai",
+             title = "OpenAI",
+             syntax = "openai:operation",
+             category = { Category.AI },
+             producerOnly = true)
+public class OpenAIEndpoint extends DefaultEndpoint {
+
+    @UriPath
+    @Metadata(required = true, description = "The operation to perform (currently only chat-completion is supported)")
+    private String operation;
+
+    @UriParam
+    private OpenAIConfiguration configuration;
+
+    private OpenAIClient client;
+
+    public OpenAIEndpoint(String uri, OpenAIComponent component, OpenAIConfiguration configuration) {
+        super(uri, component);
+        this.configuration = configuration;
+    }
+
+    @Override
+    public Producer createProducer() throws Exception {
+        if (!"chat-completion".equals(operation)) {
+            throw new IllegalArgumentException("Only 'chat-completion' operation is supported");
+        }
+        return new OpenAIProducer(this);
+    }
+
+    @Override
+    public Consumer createConsumer(Processor processor) throws Exception {
+        throw new UnsupportedOperationException("Consumer not supported for OpenAI component");
+    }
+
+    @Override
+    protected void doStart() throws Exception {
+        super.doStart();
+        client = createClient();
+    }
+
+    @Override
+    protected void doStop() throws Exception {
+        super.doStop();
+        if (client != null) {
+            client = null;
+        }
+    }
+
+    protected OpenAIClient createClient() {
+        String apiKey = resolveApiKey();
+
+        OpenAIOkHttpClient.Builder builder = OpenAIOkHttpClient.builder();
+
+        if (apiKey != null && !apiKey.isEmpty()) {
+            builder.apiKey(apiKey);
+        }
+
+        if (configuration.getBaseUrl() != null && !configuration.getBaseUrl().isEmpty()) {
+            builder.baseUrl(configuration.getBaseUrl());
+        }
+
+        return builder.build();
+    }
+
+    protected String resolveApiKey() {
+        // Priority: URI parameter > component config > environment variable > application.properties
+        if (configuration.getApiKey() != null && !configuration.getApiKey().isEmpty()) {
```

Comment:
You can use org.apache.camel.util.ObjectHelper.isNotEmpty() here for consistency